compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/SymbolFactory.java

```diff
@@ -92,7 +92,11 @@ public static Symbol getBCompiledSymbol(BSymbol symbol, String name) {
             return createTypeDefinition((BTypeSymbol) symbol, name);
         }
 
-        return null;
+        if (symbol == null) {
+            throw new IllegalArgumentException("Symbol is 'null'");
+        }
+
+        throw new IllegalArgumentException("Unsupported symbol kind: " + symbol.getClass().getName());
```

Comment:
Error message tells invalid kind, instead we show the class name. Shall we modify either the message or the input arg

Reply:
Changed kind -> type.